src/app/data-model/Decode.h

```diff
@@ -135,7 +135,10 @@ CHIP_ERROR Decode(TLV::TLVReader & reader, const ConcreteAttributePath & path, X
  *
  * Decodes an optional value (struct field, command field, event field).
  */
-template <typename X>
+template <
+    typename X,
+    typename std::enable_if_t<
+        std::is_same<decltype(Decode(std::declval<TLV::TLVReader &>(), std::declval<X &>())), CHIP_ERROR>::value, X> * = nullptr>
```

Comment:
Why is this conditional enable needed?  If there is no such Decode, the code below just won't compile, right?

Reply:
No, compiler will complain Decode is ambiguous for other types since there are more than one Decode() available.

Reply:
```
../../zzz_generated/app-common/app-common/zap-generated/cluster-objects.cpp:25470:47: error: call of overloaded 'Decode(chip::TLV::TLVReader&, chip::app::Clus
ters::ElectricalMeasurement::Attributes::ClusterRevision::TypeInfo::DecodableType&)' is ambiguous
25470 |         ReturnErrorOnFailure(DataModel::Decode(reader, clusterRevision));
      |                              ~~~~~~~~~~~~~~~~~^~~~~~~~~~~~~~~~~~~~~~~~~
../../src/lib/support/CodeUtils.h:191:23: note: in definition of macro 'ReturnErrorOnFailure'
  191 |         auto __err = (expr);                                                                                                       \
      |                       ^~~~
In file included from ../../src/app/data-model/DecodableList.h:21,
                 from ../../zzz_generated/app-common/app-common/zap-generated/cluster-objects.h:28,
                 from ../../zzz_generated/app-common/app-common/zap-generated/cluster-objects.cpp:20:
../../src/app/data-model/Decode.h:36:12: note: candidate: 'CHIP_ERROR chip::app::DataModel::Decode(chip::TLV::TLVReader&, X&) [with X = short unsigned int; ty
pename std::enable_if<std::is_integral<_Tp>::value, int>::type <anonymous> = 0; CHIP_ERROR = chip::ChipError]'
   36 | CHIP_ERROR Decode(TLV::TLVReader & reader, X & x)
      |            ^~~~~~
../../src/app/data-model/Decode.h:97:12: note: candidate: 'CHIP_ERROR chip::app::DataModel::Decode(chip::TLV::TLVReader&, X&) [with X = short unsigned int; CH
IP_ERROR = chip::ChipError]'
   97 | CHIP_ERROR Decode(TLV::TLVReader & reader, X & x)
      |            ^~~~~~
```

Reply:
I'm not talking about the Decode on line 97, which is for cluster-object structs and does need the SFINAE.

I'm talking about the ones for Nullable/Optional, which got SFINAE added when there were different-signature Decode methods involved earlier in this PR, but now that they are not involved, we don't need the SFINAE.

Concretely, I just took it out locally and everything compiles fine.  I want those 10 minutes of my life back.  ;)

Reply:
Realized that I'm deleting the wrong SFINAE, thanks.